src/Spice86.Core/Emulator/CPU/Stack.cs

```diff
@@ -3,51 +3,51 @@
 using Spice86.Core.Emulator.Memory;
 
 public class Stack {
-    private readonly Memory memory;
+    private readonly IMemory _memory;
```

Comment:
Actually with the PR I just merged, if the class only needs to access UInt8/16/32, it could depend on Indexable instead of IMemory, but I am not sure this is the purpose of your change. Did you change that to mock memory?

Reply:
Exactly.